lib/gpu/lib/src/context.dart

```diff
@@ -22,6 +24,12 @@ class GpuContext extends NativeFieldWrapperClass1 {
   @Native<Handle Function(Handle)>(
       symbol: 'InternalFlutterGpu_Context_InitializeDefault')
   external String? _initializeDefault();
+
+  /// Create a new [HostBuffer] that can be used for staging and transferring
+  /// data from the host to the GPU.
+  HostBuffer createHostBuffer() {
```

Comment:
Does this need to be on the GPUContext? All it does is call the the HostBuffer constructor, which developers could just do themselves.

Reply:
Removed for now. I added this here in the API draft because the other resource types (`Texture` and `DeviceBuffer`) need to be built with constructor methods.